packages/@apphosting/adapter-nuxt/src/utils.ts

```diff
@@ -0,0 +1,128 @@
+import type { NuxtOptions } from "@nuxt/schema";
+import fsExtra from "fs-extra";
+import { spawnSync } from "node:child_process";
+import { writeFile } from "node:fs/promises";
+import { join, normalize, posix, relative } from "node:path";
+import { stringify as yamlStringify } from "yaml";
+
+import { DEFAULT_COMMAND } from "./constants.js";
+import type { OutputBundleOptions } from "./interfaces.js";
+
+let nuxtOptions: NuxtOptions;
+export async function getConfig(cwd: string): Promise<NuxtOptions> {
+  if (!nuxtOptions) {
+    const { loadNuxtConfig } = await import("@nuxt/kit");
+    nuxtOptions = await loadNuxtConfig({ cwd });
+  }
+
+  return nuxtOptions;
+}
+
+async function getWantsBackend(cwd: string): Promise<boolean> {
+  const { ssr: wantsBackend } = await getConfig(cwd);
+
+  return wantsBackend;
+}
+
+export async function build(cwd: string, cmd = DEFAULT_COMMAND): Promise<void> {
+  const command = (await getWantsBackend(cwd)) ? "build" : "generate";
+
+  const build = spawnSync(cmd, ["run", command], {
```

Comment:
having cmd and command both be variables here is a bit confusing, could you rename one of them?

Reply:
indeed, renamed some things here to avoid confusion